client/js/widgets/widget.js

```diff
@@ -1370,27 +1374,35 @@ export class Widget extends StateManaged {
       }
 
       if(a.func == 'ROTATE') {
-        setDefaults(a, { count: 1, angle: 90, mode: 'add', collection: 'DEFAULT' });
+        setDefaults(a, { count: 1, angle: 90, mode: 'add', target: 'DEFAULT' });
         let collection;
         const mode = a.mode == 'set' ? 'to' : 'by';
         if(a.holder !== undefined) {
-          if(this.isValidID(a.holder, problems)) {
-            await w(a.holder, async holder=>{
-              for(const c of holder.children().slice(0, a.count || 999999))
-                await c.rotate(a.angle, a.mode);
-            });
+          if(collection = getCollection(a.holder)) {
+            for(const w of collections[collection]){
+              if (w.childrenTarget().length){
+                for(const c of w.childrenTarget().slice(0, a.count || 999999))
+                  await c.rotate(a.angle, a.mode);
+              } else {
+                if (w.get("type")=="seat") {
+                  problems.push(`the hand of ${w.get("id")} is empty or does not exist`);
+                } else{
+                  problems.push(`${w.get("id")} is empty or does not exist`);
```

Comment:
I don't think you can get here if `w` didn't exist as it would crash trying to call methods on `w`. It looks like `getCollection` would filter out missing widgets here: https://github.com/ArnoldSmith86/virtualtabletop/blob/6f396ceddd95a306d63edab5fe428e35b4fa035f/client/js/widgets/widget.js#L697

Reply:
You're correct this does not give the intended problem for invalid id of seats and holders. I tested only for invalid hands and empty holders. 

it does not freeze the client if the ID is invalid but the debug doesn't know that there was a problem